lib/webmock/request_pattern.rb

```diff
@@ -203,7 +203,15 @@ def matches_with_variations?(uri)
         rescue Addressable::URI::InvalidURIError
           Addressable::Template.new(@pattern.pattern)
         end
-      WebMock::Util::URI.variations_of_uri_as_strings(uri).any? { |u| template.match(u) }
+      uris = WebMock::Util::URI.variations_of_uri_as_strings(uri)
+      uris.select! { |uri| valid_uri?(uri) }
```

Comment:
@guppy0356 do you think it's worth moving `valid_uri?(uri)` inside the any block? it would be faster not to iterate over all uris, but only until the first valid and matching.